### ISSUE TITLE
Question about deepspeed load_checkpoint vs. from_pretrained

### ISSUE THREAD

**Issue by gabrielhuang** [huggingface/accelerate#994]
### System Info

```Shell
- `Accelerate` version: 0.15.0                                                                                                                                                     │·····························································································
- Platform: Linux-5.4.0-131-generic-x86_64-with-glibc2.27                                                                                                                          │·····························································································
- Python version: 3.10.8                                                                                                                                                           │·····························································································
- Numpy version: 1.24.1                                                                                                                                                            │·····························································································
- PyTorch version (GPU?): 1.13.1+cu117 (True)                                                                                                                                      │·····························································································
- `Accelerate` default config:                                                                                                                                                     │·····························································································
        - compute_environment: LOCAL_MACHINE                                                                                                                                       │·····························································································
        - distributed_type: DEEPSPEED                                                                                                                                              │·····························································································
        - mixed_precision: fp16                                                                                                                                                    │·····························································································
        - use_cpu: False                                                                                                                                                           │·····························································································
        - dynamo_backend: NO                                                                                                                                                       │·····························································································
        - num_processes: 4                                                                                                                                                         │·····························································································
        - machine_rank: 0                                                                                                                                                          │·····························································································
        - num_machines: 1                                                                                                                                                          │·····························································································
        - gpu_ids: None                                                                                                                                                            │·····························································································
        - main_process_ip: None                                                                                                                                                    │·····························································································
        - main_process_port: None                                                                                                                                                  │·····························································································
        - rdzv_backend: static                                                                                                                                                     │·····························································································
        - same_network: True                                                                                                                                                       │·····························································································
        - main_training_function: main                                                                                                                                             │·····························································································
        - deepspeed_config: {'gradient_accumulation_steps': 1, 'gradient_clipping': 1.0, 'offload_optimizer_device': 'none', 'offload_param_device': 'none', 'zero3_init_flag': Fal│·····························································································
se, 'zero_stage': 2}                                                                                                                                                               │·····························································································
        - fsdp_config: {}                                                                                                                                                          │·····························································································
        - megatron_lm_config: {}                                                                                                                                                   │·····························································································
        - downcast_bf16: no                                                                                                                                                        │·····························································································
        - tpu_name: None                                                                                                                                                           │·····························································································
        - tpu_zone: None                                                                                                                                                           │·····························································································
        - command_file: None                                                                                                                                                       │·····························································································
        - commands: None                            
```

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

I'm looking at the [official script](https://github.com/huggingface/accelerate/blob/main/examples/by_feature/deepspeed_with_config_support.py#L247) for using deepspeed stage 3 with accelerate. I just have some important questions about what model.save_checkpoint() and model.load_checkpoint (deepspeed engine methods) do:

1) Am I correct in assuming that these two functions only save/load optimizer and lr_scheduler states but not the model states, so we need to rely on .from_pretrained at the beginning of the script to resume training runs?

2) Will model.save_checkpoint() deal with loss scaling correctly?

3) Why does the tutorial use `accelerator.save_state` as well as `model.save_checkpoint`? Are both needed or is it just a leftover of the modified script?

Thank you so much.

**Comment by pacman100** [comment]
Hello @gabrielhuang, 

> 1. Am I correct in assuming that these two functions only save/load optimizer and lr_scheduler states but not the model states, so we need to rely on .from_pretrained at the beginning of the script to resume training runs?

-> Deepspeed engine methods are integrated into `accelerator.save_state` and  `accelerator.load_state`. These two functions save model, optimizer and lr_scheduler states. 

> 2. Will model.save_checkpoint() deal with loss scaling correctly?

-> What do you mean by `loss scaling correctly`, are you referring to fp16 scaling factor?

> 3. Why does the tutorial use accelerator.save_state as well as model.save_checkpoint? Are both needed or is it just a leftover of the modified script?

-> Yes, it was leftover. Please refer https://github.com/huggingface/accelerate/blob/main/src/accelerate/test_utils/scripts/external_deps/test_checkpointing.py for the latest code related to checkpointing. You can confirm how DeepSpeed checkpointing works by running following slow tests on a setup with 2 GPUs:
```bash
cd accelerate
export RUN_SLOW=true
python -m pytest -s -v ./tests/deepspeed -k test_checkpointing
```

**Comment by gabrielhuang** [comment]
1. Just to be clear, `model.load_checkpoint()` will load or not load the model weights?
2. Yes, i mean the thing that prints "OVERFLOW!" Rank 0 Skipping step. etc... speaking of which, I would like to confirm how I can set the `initial_dynamic_scale`? it defaults to 4294967296 for me, even when i pass a `initial_scale_power` of 15 using the `--config_file` flag:
```yaml
deepspeed_config:
  gradient_accumulation_steps: 1
  gradient_clipping: 1.0
  zero_stage: 3
  offload_optimizer_device: none
  offload_param_device: none
  zero3_init_flag: true
  fp16:
    initial_scale_power: 15  # max loss of 2.0
  zero3_save_16bit_model: true  # necessary for saving checkpoints correctly
```
3. Thanks for the tip of checking the test code, I hadn't thought of that.
